sklearn/cluster/tests/test_k_means.py

```diff
@@ -259,10 +235,12 @@ def test_k_means_random_init_not_precomputed():
     _check_fitted_model(km)
 
 
-def test_k_means_perfect_init():
-    km = KMeans(init=centers.copy(), n_clusters=n_clusters, random_state=42,
-                n_init=1)
-    km.fit(X)
+@pytest.mark.parametrize('representation', ['dense', 'sparse'])
```

Comment:
It's just for the readability when you run pytest.

With your proposition tests will appear as 
`test_whatever_test_name[data0]`
`test_whatever_test_name[data1]`

Here it will appear as 
`test_whatever_test_name[dense]`
`test_whatever_test_name[sparse]`

I just find it easier to track which parameters make some test fail.

Reply:
Yeah that's a good point

Reply:
FYI, it's possible to provide `ids` as a workaround,
```py
@pytest.mark.parametrize('data', (X, Xcsr), ids=('dense', 'sparse'))
```
maybe that's a bit more direct?

Reply:
I didn't know that. This is better ! Thanks